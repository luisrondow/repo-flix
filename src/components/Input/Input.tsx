import { forwardRef } from 'react'
import { InputProps } from './Input.types'

import Text from '../Text'

const Input = forwardRef<HTMLInputElement, InputProps>((props, ref) => {
  const { label, placeholder, type, value, defaultValue, onChange, className, error, ...rest } =
    props

  return (
    <div className={className}>
      <label>
        <Text as="h1" className="mb-2 text-base text-white">
          {label}
        </Text>
        <input
          ref={ref}
          type={type}
          placeholder={placeholder}
          value={value}
          onChange={onChange}
          className="h-12 w-full rounded-md bg-base px-4 text-white"
          defaultValue={defaultValue}
          {...rest}
        />
        <Text className="mt-2 text-sm font-bold text-red">{error}</Text>
      </label>
    </div>
  )
})

Input.displayName = 'Input'

export default Input
