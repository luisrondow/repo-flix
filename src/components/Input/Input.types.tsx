export type InputProps = {
  label: string
  placeholder: string
  type: string
  value: string
  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void
  className?: string
}
