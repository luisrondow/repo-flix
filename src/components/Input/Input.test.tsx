import { render } from '@testing-library/react'
import Input from './Input'

describe('Input', () => {
  it('should render correctly', () => {
    const { container, getByLabelText } = render(
      <Input label="Mock label" placeholder="Mock placeholder" type="text" error="" />,
    )

    expect(getByLabelText('Mock label')).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })

  it('should render correctly when error is not empty', () => {
    const { container, getByText } = render(
      <Input label="Mock label" placeholder="Mock placeholder" type="text" error="Mock error" />,
    )

    expect(getByText('Mock error')).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })

  it('should render correctly when type is password', () => {
    const { container, getByLabelText } = render(
      <Input label="Mock label" placeholder="Mock placeholder" type="password" error="" />,
    )

    expect(getByLabelText('Mock label')).toHaveAttribute('type', 'password')
    expect(container).toMatchSnapshot()
  })
})
