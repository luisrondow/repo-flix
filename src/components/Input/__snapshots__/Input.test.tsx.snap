// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render correctly 1`] = `
<div>
  <div>
    <label>
      <h1
        class="mb-2 text-base text-white text-[#e0def4] font-roboto text-2xl font-bold sm:text-4xl"
      >
        Mock label
      </h1>
      <input
        class="h-12 w-full rounded-md bg-base px-4 text-white"
        placeholder="Mock placeholder"
        type="text"
        value=""
      />
      <p
        class="mt-2 text-sm font-bold text-red text-[#e0def4] font-roboto"
      />
    </label>
  </div>
</div>
`;

exports[`Input should render correctly when error is not empty 1`] = `
<div>
  <div>
    <label>
      <h1
        class="mb-2 text-base text-white text-[#e0def4] font-roboto text-2xl font-bold sm:text-4xl"
      >
        Mock label
      </h1>
      <input
        class="h-12 w-full rounded-md bg-base px-4 text-white"
        placeholder="Mock placeholder"
        type="text"
        value=""
      />
      <p
        class="mt-2 text-sm font-bold text-red text-[#e0def4] font-roboto"
      >
        Mock error
      </p>
    </label>
  </div>
</div>
`;

exports[`Input should render correctly when type is password 1`] = `
<div>
  <div>
    <label>
      <h1
        class="mb-2 text-base text-white text-[#e0def4] font-roboto text-2xl font-bold sm:text-4xl"
      >
        Mock label
      </h1>
      <input
        class="h-12 w-full rounded-md bg-base px-4 text-white"
        placeholder="Mock placeholder"
        type="password"
        value=""
      />
      <p
        class="mt-2 text-sm font-bold text-red text-[#e0def4] font-roboto"
      />
    </label>
  </div>
</div>
`;
